examples/embeddable_examples/public/saved_object_embeddable/so_embeddable.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Subscription } from 'rxjs';
+import { CoreStart } from 'kibana/public';
+import {
+  Embeddable,
+  EmbeddableInput,
+  IContainer,
+  EmbeddableOutput,
+} from '../../../../src/plugins/embeddable/public';
+import { SOComponent } from './so_component';
+
+export const SO_EMBEDDABLE = 'SO_EMBEDDABLE';
+
+export interface SOInput extends EmbeddableInput {
+  id: string;
+  type: string;
+}
+
+interface Config {
+  input: SOInput;
+  parent?: IContainer;
+  savedObjectClient: CoreStart['savedObjects']['client'];
+}
+
+export class SOEmbeddable extends Embeddable<SOInput, EmbeddableOutput> {
+  // The type of this embeddable. This will be used to find the appropriate factory
+  // to instantiate this kind of embeddable.
+  public readonly type = SO_EMBEDDABLE;
+  private subscription: Subscription;
+  private node?: HTMLElement;
+  private savedObjectClient: CoreStart['savedObjects']['client'];
+
+  constructor({ input, parent, savedObjectClient }: Config) {
+    super(input, { references: [{ id: input.id, type: input.type, name: '' }] }, parent);
```

Comment:
this is a bug, we need to get all references via `const object = await this.savedObjectClient.get(this.input.type, this.input.id);` but then I can't pass it through to `super`, so it needs to be in factory.